[rocprofiler-sdk] rename librocprofv3-attach.so

meserve-amd opened the pull request:
## Motivation

In anticipation of formalizing an API for custom tools to attach with, we will rename librocprofv3-attach.so to librocprofiler-sdk-rocattach.so.  This name was chosen because the library is technically not part of rocprofv3. We also anticipate formalizing and publishing a public API with this library, which we are calling ROCAttach. This API will be used by custom tools who wish to create their own attachment scenarios, so they do not need to learn the inner workings of ptrace or the protocol used to communicate with rocprofiler-sdk in the running process.

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->

## Technical Details

This PR is staged over a documentation change that I expect to be merged first.

I've kept just the rename separate for easier review and gitfu. Another review and documentation step for this API is to follow.

<!-- Explain the changes along with any relevant GitHub links. -->

## Test Plan

Use existing CI

<!-- Explain any relevant testing done to verify this PR. -->

## Test Result

First result - failed due to incorrect naming of .so file, fixed this
Second result - some existing failures, need to fix before validating this change is ok

<!-- Briefly summarize test outcomes. -->

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

meserve-amd committed:
attach: rename librocprofv3-attach

- Renames library to librocprofiler-sdk-rocattach
- ROCAttach library will be formalized and documented in future commit

itrowbri reviewed (approved):
I'm not familiar with the component sections of the code, but the changes seem simple enough. Will we need to add some tests outside of the `tests/rocprofv3` section if we formalize this change given that attachment won't be limited to rocprofv3 anymore?

vedithal-amd commented:
Please rename the library hardcoded in rocprofiler-compute at https://github.com/ROCm/rocm-systems/blob/develop/projects/rocprofiler-compute/src/rocprof_compute_profile/profiler_rocprofiler_sdk.py

jrmadsen commented on projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-rocattach/details/filesystem.hpp:
This file shouldn't exist and the preprocessor defines are wrong. Use the one in `lib/common/filesystem.hpp`

jrmadsen commented on projects/rocprofiler-sdk/source/bin/rocprof-attach.py:
This file should probably be named `rocprof-attach`

jrmadsen commented on projects/rocprofiler-sdk/source/bin/rocprofv3-attach.py:
```diff
@@ -30,7 +30,7 @@
 
 ROCPROFV3_ATTACH_DIR = os.path.dirname(os.path.realpath(__file__))
```
This variable name needs updating

jrmadsen commented on projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-rocattach/rocattach.cpp:
If the purpose is make an API, why are there no headers?

meserve-amd replied:
Those will be in another PR along with all other changes to make this a public API. I wanted to get the rename done ASAP.

meserve-amd replied:
This is contained in #1653

meserve-amd committed:
Address review comments

- Rename rocprofv3-attach.py to rocprof-attach.py
- Use common filesystem.hpp in rocattach

bwelton commented on projects/rocprofiler-sdk/cmake/rocprofiler_config_packaging.cmake:
```diff
@@ -78,26 +80,32 @@ set(COMPONENT_NAME_tests "rocprofiler-sdk-tests")
 set(COMPONENT_NAME_roctx "rocprofiler-sdk-roctx")
 set(COMPONENT_NAME_rocpd "rocprofiler-sdk-rocpd")
 set(COMPONENT_NAME_benchmark "rocprofiler-sdk-benchmark")
+set(COMPONENT_NAME_rocattach "rocprofiler-sdk-rocattch")
```
Small typo, rocattach instead of attch.

bwelton reviewed (changes requested):
Fix the typo, may fix the tests as well.

meserve-amd committed:
Fix component name typo

jrmadsen commented on projects/rocprofiler-sdk/source/bin/rocprof-attach.py:
Why does this script rely solely on environment variables? Why not create argparse support?

meserve-amd replied:
I agree with this comment, but I think that belongs in a new PR. Users can use this as an import, but it should be directly callable as well without environment variables  I will add that to the rocattach support or a different PR.

meserve-amd replied:
This is contained in #1653

jrmadsen commented on projects/rocprofiler-sdk/source/docs/api-reference/process_attachment.rst:
```diff
@@ -16,7 +16,7 @@ This document provides the technical details needed to implement a process attac
 ===================================
 
-The python file ``rocprofv3-attach.py`` defines a main function that can be used for attachment:
+The python file ``rocprof-attach.py`` defines a main function that can be used for attachment:
```
From the user POV, the executable does not have a “.py” extension.

meserve-amd replied:
Fixed

jrmadsen commented on projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-rocattach/ptrace_session.cpp:
```diff
@@ -48,6 +48,8 @@ static_assert(sizeof(void*) == 8);
 #endif
 
+namespace fs = rocprofiler::common::filesystem;
+
```
All the other code in the repo does not use a bare anonymous namespace. See [here](https://github.com/ROCm/rocm-systems/blob/24cb8c4deb9f5fe1de85b014699afadd6a3a22b3/projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-roctx/roctx.cpp#L44)

meserve-amd replied:
Ok, I will do this, but I will add it into the next PR for ROCAttach since I am already making extensive changes there and would rather not rebase.

meserve-amd replied:
This is contained in #1653

meserve-amd committed:
Doc fixup

JeniferC99 committed:
Merge branch 'develop' into users/meserve-amd/attach/rocattach-rename

JeniferC99 commented:
psdb passing rocprofiler test -- override
